idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -827,12 +831,30 @@ def build(self):
                                 f"contained unrecognised phase {p}."
                             )
                         else:
-                            try:
-                                meth.build_parameters(cobj, p)
-                            except AttributeError:
-                                # No build_parameters method
-                                # Assume it is not needed and continue
-                                continue
+                            if hasattr(meth, "build_parameters"):
+                                build_parameters = meth.build_parameters
+                            else:
+                                # If not, guess meth is a class holding property subclasses
+                                try:
```

Comment:
Just to be sure, could you explain why this was needed (as a comment on the PR, not in the code)?

Reply:
The way `diffus_phase_comp` is implemented now, `build_parameters` is a method directly on the parent class. However, in order to support providing `visc_d_phase_comp` and `therm_cond_phase_comp` methods through the `ConstantProperty` class, we need to also support looking up subclasses with the name, e.g., `visc_d_phase_comp` on the parent class.

Reply:
OK, I think that is a good enough reason. We could insist they provide the sub-class from ConstantProperties, but I don't see any reason not to support this API either (I can also see some good reasons to do it to avoid unintended conflicts, e.g. getting cp from one module but enthalpy from another).